src/validations/rules/sap.sch

```diff
@@ -171,6 +193,70 @@
 
     </sch:pattern>
 
+    <sch:pattern
+        id="control-selection">
+        <sch:let
+            name="ssp-controls"
+            value="$ssp-doc//oscal:implemented-requirement/@control-id" />
+        <sch:rule
+            context="oscal:control-selection">
+            <sch:let
+                name="exclude-control-ids"
+                value="oscal:exclude-control/@control-id ! xs:string(.)" />
+            <sch:let
+                name="include-control-ids"
+                value="oscal:include-control/@control-id ! xs:string(.)" />
+            <sch:let
+                name="matching-control-ids"
+                value="$exclude-control-ids[. = $include-control-ids]" />
+            <sch:assert
+                diagnostics="include-all-or-include-control-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP) §4.1"
+                id="include-all-or-include-control"
+                role="fatal"
+                test="(oscal:include-all and not(oscal:include-control)) or (oscal:include-control and not(oscal:include-all))">An OSCAL SAP control
+                selection elements must have either an include-all element or include-control element(s) children.</sch:assert>
+            <sch:assert
+                diagnostics="duplicate-exclude-control-and-include-control-values-diagnostic"
+                id="duplicate-exclude-control-and-include-control-values"
+                role="error"
+                test="count($matching-control-ids) = 0">The exclude-control and include-control sibling element @control-id values must be
+                different.</sch:assert>
+            <sch:assert
+                diagnostics="duplicate-include-control-values-diagnostic"
+                id="duplicate-include-control-values"
+                role="error"
+                test="count($include-control-ids) eq count(distinct-values($include-control-ids))">The include-control/@control-id values must be
+                different.</sch:assert>
```

Comment:
Suggest "The include-control/@control-id values must not be duplicated." or equivalent.

Reply:
Sounds good.  Changed.